crates/fuel-core/src/state.rs

```diff
@@ -157,4 +159,19 @@ where
             core::iter::empty().into_boxed()
         }
     }
+
+    fn iter_store_keys(
+        &self,
+        column: Self::Column,
+        prefix: Option<&[u8]>,
+        start: Option<&[u8]>,
+        direction: IterDirection,
+    ) -> BoxedIter<fuel_core_storage::kv_store::KeyItem> {
+        // We cannot define iter_store_keys appropriately for the `ChangesIterator`,
+        // because we have to filter out the keys that were removed, which are
+        // marked as `WriteOperation::Remove` in the value
+        self.iter_store(column, prefix, start, direction)
+            .map(|item| item.map(|(key, _)| key))
+            .into_boxed()
```

Comment:
You could copy/paste everything from the function above and just remove `value.clone()`=)

Reply:
sure, addressed in 06bd0e962
thanks